test/e2e/case37_no_name_test.go

```diff
@@ -0,0 +1,372 @@
+			Expect(kind).Should(Equal("Ingress"))
+
+			By("Check the reasons of relatedObjects")
+			relatedObjectsOne := relatedObjects[0].(map[string]interface{})
+			Expect(relatedObjectsOne["reason"].(string)).Should(Equal("Resource found but does not match"))
+
+			By("Check the name of relatedObject is *")
+			name, _, err := unstructured.NestedString(relatedObjects[0].(map[string]interface{}),
+				"object", "metadata", "name")
+			Expect(err).ShouldNot(HaveOccurred())
+			Expect(name).Should(Equal("*"))
+		})
+		It("should have 1 Compliant relatedObject under the policy's status", func() {
+			By("Apply good ingress")
+			utils.Kubectl("apply", "-f", case37GoodIngressPath)
+
+			var managedPlc *unstructured.Unstructured
+
+			Eventually(func() interface{} {
+				managedPlc = utils.GetWithTimeout(clientManagedDynamic, gvrConfigPolicy,
+					case37PolicyNSName, testNamespace, true, defaultTimeoutSeconds)
+
+				return utils.GetComplianceState(managedPlc)
+			}, defaultTimeoutSeconds, 1).Should(Equal("Compliant"))
+
+			var relatedObjects []interface{}
+
+			Eventually(func(g Gomega) interface{} {
+				managedPlc = utils.GetWithTimeout(clientManagedDynamic, gvrConfigPolicy,
+					case37PolicyNSName, testNamespace, true, defaultTimeoutSeconds)
+
+				var err error
+
+				relatedObjects, _, err = unstructured.NestedSlice(managedPlc.Object, "status", "relatedObjects")
+
+				g.Expect(err).ShouldNot(HaveOccurred())
+				g.Expect(relatedObjects).ShouldNot(BeNil())
+
+				return relatedObjects
+			}, defaultTimeoutSeconds*2, 1).Should(HaveLen(1))
+
+			//  NonCompliant relatedObjects Should be deleted and only the Compliant relatedObject remain
+			By("Check the kind of relatedObjects")
+			kind := relatedObjects[0].(map[string]interface{})["object"].(map[string]interface{})["kind"].(string)
+			Expect(kind).Should(Equal("Ingress"))
+
+			By("Check the reason of relatedObjects")
+			reason := relatedObjects[0].(map[string]interface{})["reason"].(string)
+			Expect(reason).Should(Equal("Resource found as expected"))
+		})
+		AfterAll(func() {
+			utils.Kubectl("delete", "-f", case37PolicyNSPath, "-n", testNamespace, "--ignore-not-found")
+			utils.Kubectl("delete", "-f", case37GoodIngressPath, "--ignore-not-found")
+			utils.Kubectl("delete", "-f", case37BadIngressPath, "--ignore-not-found")
+			utils.GetWithTimeout(clientManagedDynamic, gvrConfigPolicy,
+				case37PolicyNSName, testNamespace, false, defaultTimeoutSeconds)
+		})
+	})
+	Describe("Test a mustnothave", Ordered, func() {
+		BeforeAll(func() {
+			By("Creating a policy on managed")
+			utils.Kubectl("apply", "-f", case37PolicyNotHavePath, "-n", testNamespace)
+			plc := utils.GetWithTimeout(clientManagedDynamic, gvrConfigPolicy,
+				case37PolicyMustnothaveName, testNamespace, true, defaultTimeoutSeconds)
+			Expect(plc).ShouldNot(BeNil())
+		})
+		It("should have 0 Compliant relatedObject under the policy's status", func() {
```

Comment:
I don't think this is correctly describing the test since it's expecting related objects to have a length of 1. Could you please reword it?